lib/msal-browser/docs/shr-client-claims.md

```diff
@@ -0,0 +1,48 @@
+# Custom SHR Claims
+
+As an enhancement of Access Token Proof-of-Possesion, MSAL Browser provides a way to insert custom client claims into a `Signed HTTP Request`, also known as a `PoP Token`. These custom SHR claims can be added to any token request that uses the `POP` authentication scheme.
+
+Given that the payload of a `Signed HTTP Request` has a specific format, rather than allowing custom claim names which could collide with any of the [first-order SHR claims](https://tools.ietf.org/html/draft-ietf-oauth-signed-http-request-03#section-3), MSAL Browser allows custom client claims to be passed in as a string, which will be added to the Signed HTTP Request `JWT` as the value of a claim called `client_claims`. This string will usually be a serialized object containing the custom claims, but MSAL will not parse or verify the string in any way.
+
+Given that MSAL does not cache `Signed HTTP Requests` (the access token secret is cached and the SHR payload is built and signed just-in-time whenever `acquireTokenSilent` is called), custom client claims will **not** be cached either. This means that the claims must also be passed into every `acquireTokenSilent` call in order for them to be added to the resulting `SignedHTTPRequest`.
```

Comment:
Does passing client claims automatically result in a cache miss like passing `claims` does?

Reply:
No, the client claims aren't taken into account for the caching logic. They go straight into the SHR "envelope".